### DISCUSSION TITLE
Introduce a "next" branch and docker image tag

### DISCUSSION THREAD

**Discussion by asoorm** [TykTechnologies/tyk-operator#244] [Ideas] [answered]
Users getting started with / trialing / helping us test Tyk Operator are likely to git clone this repository from the master branch. They are likely not going to checkout to a tag to deploy a specific version of the Operator.

The Master branch is getting merged to quite regularly. Whilst the master branch should be of production quality, the docker image is not built from the master branch, the docker image `latest` is built from the latest tag.

This means that there can be a mismatch between master in git and latest inside dockerhub. This could possibly negatively impact those who installed CRDs from Latest, but have a version of the operator installed which does not match the CRD. It might also negatively impact trialists and those who are actively working with our latest feature set.

As such, this proposal is to introduce a `next` or `develop` branch. By adopting a kind of git-flow workflow, where all PRs get merged into the next branch, we can ensure that `latest` tag in DockerHub always matches the master branch inside GitHub. Only when we cut a release will the master branch become updated. Further, whenever we merge into `next`, we can always build and tag a `tyk-operator:next` docker image, allowing team members to easily test upcoming functionality, without impacting real users.

@gernest @excieve - thoughts please.

**Comment by gernest** [comment, marked as answer]
> the docker image latest is built from the latest tag

Maybe this should be the focus instead? Can we make docket build `latest` from master branch?
